### ISSUE TITLE
Unmarshaling unknown values is not supported

### ISSUE THREAD

**Issue by flvndh** [hashicorp/terraform-provider-kubernetes#1402]
### Terraform Version, Provider Version and Kubernetes Version
```
> tf version
Terraform v1.0.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/kubernetes v2.4.1
+ provider registry.terraform.io/hashicorp/template v2.2.0
```

### Affected Resource(s)
- `kubernetes_manifest`

### Terraform Configuration Files
```hcl
resource "kubernetes_namespace" "this" {
  metadata {
    name = var.kubernetes_namespace
  }
}

data "template_file" "persistent_volume_manifest" {
  template = file("assets/persistent-volume.yaml")
  vars = {
    name      = var.persistent_volume_name
    host_path = var.host_path
  }
}

resource "kubernetes_manifest" "persistent_volume" {
  manifest = yamldecode(data.template_file.persistent_volume_manifest.rendered)
}

data "template_file" "persistent_volume_claim_manifest" {
  template = file("assets/persistent-volume-claim.yaml")
  vars = {
    name        = var.persistent_volume_claim_name
    namespace   = kubernetes_namespace.this.metadata.0.name
    volume_name = kubernetes_manifest.persistent_volume.manifest.metadata.name
  }
}

resource "kubernetes_manifest" "persistent_volume_claim" {
  manifest = yamldecode(data.template_file.persistent_volume_claim_manifest.rendered)
}
```

```yaml
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: ${name}
  namespace: ${namespace}
spec:
  resources:
    requests:
      storage: 5Gi
  accessModes:
    - ReadOnlyMany
  storageClassName: hostpath
  volumeName: ${volume_name}

```

### Debug Output
```
> tf plan
╷
│ Error: Failed to determine GroupVersionResource for manifest
│ 
│   with kubernetes_manifest.persistent_volume_claim,
│   on main.tf line 28, in resource "kubernetes_manifest" "persistent_volume_claim":
│   28: resource "kubernetes_manifest" "persistent_volume_claim" {
│ 
│ unmarshaling unknown values is not supported
```

### Steps to Reproduce
1. `terraform plan`

**Comment by alexsomesan** [comment]
TL;DR; The template_file data-source [is deprecated](https://registry.terraform.io/providers/hashicorp/template/latest/docs/data-sources/file) and its use is discouraged. It's also the cause of your issues in this case.
Please use the built-in `templatefile()` function instead of this datasource. For example:
```
locals {
  persistent_volume_claim_manifest = templatefile("persistent-volume-claim.yaml",
  {
    name      = var.persistent_volume_claim_name
    namespace = kubernetes_namespace.this.metadata.0.name
    volume_name = kubernetes_manifest.persistent_volume.manifest.metadata.name
  })
}
```

The issue you are seeing here is caused by the value of the template data-source not being evaluated early enough for the `kubernetes_manifest` resource to consume it successfully. Unlike other resource, the kubernetes_manifest requires some parts of its configuration during the planning phase in order to establish the resource type. However, the output of the data-source is not yet fully defined at this point due to it's dependencies on the input values and the way it's evaluated.

**Comment by flvndh** [comment]
My bad 🙈 Thanks @alexsomesan for the answer 😉
